base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1179 @@
+/*
+ * PROJECT:         ReactOS Disk Cleanup
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Utility functions
+ * COPYRIGHT:       Copyright 2020 Arnav Bhatt (arnavbhatt288 at gmail dot com)
+ */
+ 
+#include "resource.h"
+#include "precomp.h"
+
+#define KB 1024
+#define MB 1024 * KB
+#define GB 1024 * MB
+#define TB 1024 * GB
+
+#define ALLOW_FILE_REMOVAL 2
+#define DISALLOW_FILE_REMOVAL 3
+
+#define CX_BITMAP 20
+#define CY_BITMAP 20
+
+typedef enum
+{
+    OLD_CHKDSK_FILES = 0,
+    RAPPS_FILES = 1,
+    RECYCLE_BIN = 2,
+    TEMPORARY_FILE = 3
+} DIRECTORIES;
+
+#define ETDT_DISABLE        0x00000001
+#define ETDT_ENABLE         0x00000002
+#define ETDT_USETABTEXTURE  0x00000004
+#define ETDT_ENABLETAB      (ETDT_ENABLE  | ETDT_USETABTEXTURE)
+
+typedef HRESULT (WINAPI * ETDTProc) (HWND, DWORD);
+
+DWORD WINAPI SizeCheck(LPVOID lpParam)
+{
+    HKEY hRegKey = NULL;
+    //SETTINGS_INFO SettingsInfo;
+    //ZeroMemory(&SettingsInfo, sizeof(SETTINGS_INFO));
+    //DWORD dwSize = sizeof(SettingsInfo);
+    
+    if (RegOpenKeyExW(HKEY_CURRENT_USER,
+                      L"Software\\ReactOS\\rapps",
+                      0,
+                      KEY_QUERY_VALUE,
+                      &hRegKey) != ERROR_SUCCESS)
+    {
+        DPRINT("RegOpenKeyExW(): Failed to open a registry key!\n");
+    }
+    
+    HWND hwnd = (HWND)lpParam;
+    
+    HWND hProgressBar = NULL;
+    WCHAR TargetedDir[MAX_PATH] = { 0 };
+    SHQUERYRBINFO RecycleBinInfo;
+    ZeroMemory(&RecycleBinInfo, sizeof(RecycleBinInfo));
+    RecycleBinInfo.cbSize = sizeof(RecycleBinInfo);
+
+    hProgressBar = GetDlgItem(hwnd, IDC_PROGRESS_1);
+    
+    if(bv.SysDrive == TRUE)
```

Comment:
Nit: `== TRUE` is generally discouraged, because `BOOL` isn't a true boolean type and could potentially have values like `2` (which should be considered true but don't have the value `TRUE`).